### PR TITLE
Implement Fenwick Tree data structures with tests and documentation

### PR DESCRIPTION
**Pull request by lrleon**
## Summary by CodeRabbit

* **New Features**
  * Added three Fenwick tree variants and a multi-scenario example demonstrating order book, intraday P&L, and promotions.

* **Tests**
  * Added extensive unit and stress tests covering construction, updates, queries, find-kth, range semantics, numeric/generic behaviors, and edge cases.

* **Documentation**
  * Expanded English and Spanish READMEs with usage examples, API notes, and complexity guidance.

* **Chores**
  * Registered new header and example/test targets; added coverage tooling, CI helpers, and a pre-merge doc-coverage check; improved test-run command handling.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -595,6 +601,175 @@ int main() {
...
+┌────────────────────────────────────────────────────────────────────────────┐
+│                         FENWICK TREE VARIANTS                              │
+├──────────────────┬──────────────┬──────────────┬──────────────┬────────────┤
+│     Operation    │ Gen (Abelian)│ Fenwick_Tree │  Range_Fen   │  Complexity │
+├──────────────────┼──────────────┼──────────────┼──────────────┼────────────┤
+│ point_update     │     O(log n) │   O(log n)   │   O(log n)   │            │
+│ prefix / get     │     O(log n) │   O(log n)   │   O(log n)   │            │
+│ range_query      │     O(log n) │   O(log n)   │   O(log n)   │            │
+│ range_update     │       N/A    │     N/A      │   O(log n)   │            │
+│ find_kth         │       N/A    │   O(log n)   │     N/A      │            │
```
[the remark this answers is not in the document]
Comment (lrleon) [reply README.md:618]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/lrleon/Aleph-w/pull/27#discussion_r2780393570)

**Comment by lrleon** [comment]
@coderabbitai review

**Comment by lrleon** [comment]
@coderabbitai review

**Comment by lrleon** [comment]
@codex review

**Comment by lrleon** [comment]
@coderabbitai review

**Comment by lrleon** [comment]
@coderabbitai resume
